planner/core/exhaust_physical_plans.go

```diff
@@ -2038,6 +2038,13 @@ func (p *LogicalProjection) exhaustPhysicalPlans(prop *property.PhysicalProperty
 		mppProp.TaskTp = property.MppTaskType
 		newProps = append(newProps, mppProp)
 	}
+	if newProp.TaskTp != property.CopSingleReadTaskType && p.SCtx().GetSessionVars().AllowProjectionPushDown && p.canPushToCop(kv.TiKV) &&
```

Comment:
add a TODO to merge pushing down projection for both tiflash and tikv?

Reply:
it supports tiflash now.